app/main.py

```diff
@@ -1,2 +1,64 @@
+from typing import Any
+
+
+class Node:
+    def __init__(self, key: Any, value: Any, hash_value: int) -> None:
+        self.key = key
+        self.value = value
+        self.hash = hash_value
+        self.next = None
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.size = 0
+        self.load_factor = 0.7
+        self.table = [None] * self.capacity
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        current_index = self._hash_index(key)
+        current = self.table[current_index]
+        while current is not None:
+            if current.key == key:
+                current.value = value
+                return
+            current = current.next
+
+        new_node = Node(key, value, hash(key))
+        new_node.next = self.table[current_index]
+        self.table[current_index] = new_node
+        self.size += 1
+
+        if self.size / self.capacity > self.load_factor:
+            self._resize()
+
+    def __getitem__(self, key: Any) -> None:
```

Comment:
The return type for the __getitem__ method should be 'Any' instead of 'None', as it returns the value associated with the key.